### PR TITLE
CORGI-195: Fix corrupt data

### PR DESCRIPTION
**Pull request by juspence**
@RedHatProductSecurity/corgi-devs EDITED after some tweaking. The django-mptt PR mentioned in CORGI-195 does fix the bug with concurrent writes corrupting the tree hierarchies. I've forked the django-mptt library, rebased and merged that PR, and tested locally.

I did try just adding a new migration that switched tree_ids from ints to UUIDs, but code started raising errors. So I had to alter the first Corgi migration + wipe out and reload my local data to make this work. But we would have needed to wipe out data anyway, since at least some of it was already corrupted, and rebuilding that many nodes (in stage / prod) will take an unreasonable amount of time.

### CODE REVIEW

**Comment by JimFuller-RedHat** [comment]
I think this also applicable for CORGI-191

File: corgi/core/models.py [hunk 1]
```diff
@@ -839,30 +840,25 @@ def get_roots(self) -> list[ComponentNode]:
...
+            if root.obj.type == Component.Type.CONTAINER_IMAGE:
+                # TODO if we change the CONTAINER->RPM ComponentNode.type to something besides
+                # 'PROVIDES' we would check for that type here to prevent 'hardcoding' the
+                # container -> rpm relationship here.
+                # RPMs are included as children of Containers as well as SRPMs
+                # We don't want to include Containers in the RPMs roots.
+                # Partly because RPMs in containers can have unprocesses SRPMs
+                # And partly because we use roots to find upstream components,
+                # and it's not true to say that rpms share upstreams with containers
+                rpm_desendant = False
```
Comment (mprpic) [inline corgi/core/models.py]: I know this is unrelated to your change, but could you fix this typo please? :smiley:

**Review by mprpic — APPROVED** [review]
LGTM!

**Comment by JimFuller-RedHat** [comment]
this PR has caused some instability, likely I will revert it

**Pull request merged**